### ISSUE TITLE
Schematics... 

### ISSUE THREAD

**Issue by dixi83** [bigtreetech/Rumba32#1]
Please supply them is you want to sell this board...

**Comment by ellensp** [comment]
Please provide schematics.

**Comment by Mekanikalistik** [comment]
Where is the shematics of this board ?

In marlin code there is a line about SERVO PINS


#if MB(RUMBA32_V1_1)

  #define SERVO0_PIN                        PA15
  #undef BTN_PIN

Where is this pin located on board ?

**Comment by FinnsMax** [comment]
PA15 - TP4

**Comment by Mekanikalistik** [comment]
exactly which point on the board ?

**Comment by FinnsMax** [comment]
TP4, test paint

**Comment by FinnsMax** [comment]
https://github.com/makerbase-mks/MKS-RUMBA32/blob/master/hardware/MKS%20RUMBA32%20V1.0_002/MKS%20RUMBA32%20V1.0_002%20TOP.pdf

**Comment by whiterabbit62** [comment]
Does MKS RUMB32 have same pinout as BTT rumba32 ?

**Comment by Mekanikalistik** [comment]
Does MKS RUMB32 have same pinout with BTT rumba32 ?

**Comment by FinnsMax** [comment]
yes

**Comment by Mekanikalistik** [comment]
Thank you so much :)

**Comment by bagobor** [comment]
@FinnsMax I suppose all bugs are here as well? ex: https://github.com/makerbase-mks/MKS-RUMBA32/issues/2

**Comment by FinnsMax** [comment]
I suppose all bugs are here as well? ex: lack of filters at endstops and thermistor inputs

**Comment by dixi83** [comment]
It is not the same as the MKS it has many comparable things, for example to get uart drives (like TMC 2208/9) to work, this is what i found after tracing back the board:
`pins_RUMBA32_AUS3D.h`  (which is the pin file BTT uses in there example, the MKS pin file is not even used):
```
    #define X_SERIAL_TX_PIN                 PC14 //PA14 BTT Rumba32 only uses 1 pin for uart
    #define X_SERIAL_RX_PIN                 PC14

    #define Y_SERIAL_TX_PIN                 PE4 //PA13
    #define Y_SERIAL_RX_PIN                 PE4

    #define Z_SERIAL_TX_PIN                 PE0 //PB10
    #define Z_SERIAL_RX_PIN                 PE0

    #define E0_SERIAL_TX_PIN                PC13 //PD11
    #define E0_SERIAL_RX_PIN                PC13

    #define E1_SERIAL_TX_PIN                PD5 //PB3
    #define E1_SERIAL_RX_PIN                PD5

    #define E2_SERIAL_TX_PIN                PD1 //PB4
    #define E2_SERIAL_RX_PIN                PD1
```
And I have found more differences and undocumented things. I'll keep this for my self until the lazy company called "bigtreetech" supplied there schematics or unless someone specifically ask for it here. Because the word "support" is not in there dictionary.

**Comment by dixi83** [comment]
> #if MB(RUMBA32_V1_1)
> 
> #define SERVO0_PIN PA15
> #undef BTN_PIN
> 
> Where is this pin located on board ?

to answer a serious question instead of suppling wrong information,
PA15 is not used or wired out on the board, it is located here (the yellow spot near my arrow):
![image](https://user-images.githubusercontent.com/10693808/110244414-e0cb3800-7f5e-11eb-9a35-bdfb706828db.png)

**Comment by Mekanikalistik** [comment]
Yes, i also checked their marlin software to find correct pins. What are other things you came up with this board.. i have 2 of them.. they are all working fine. Just dont forget to pull down or pull up enstop pins depending on your setup. I am having problems with IR probe and seems like it is solved after i pulled down the Zmin pin (STM32f4 supports internal pull down too)

Currently , i am trying to solve the noise/loud 0.9degree steppers +mks2209 v.2 +24V volt setup and there seems no solution.

**Comment by dixi83** [comment]
I'm using BLtouch for which I redefined the Z-Max pin as servo, and use the Z-Min pin as probe pin. pin configuration now looks like this:
`pins_RUMBA32_common.h` : 
```
#define Z_MAX_PIN                           PA15 // PA15 is not connected to anything
```

`pins_RUMBA32_AUS3D.h` :
```
#define SERVO0_PIN                          PD10 // PD10 was Z_MAX_PIN 
```

I am using TMC2209 without any issues now I had problems with 1 drive, which was not running at al but only making noise, I found out this drive was wired different than the other drives which came from the same order... so I swopped 2 wires and it ran.

**Edit:** mistake about the pin number PB10 should be PD10 of course...

**Comment by Mekanikalistik** [comment]
> 
> 
> I'm using BLtouch for which I redefined the Z-Max pin as servo, and use the Z-Min pin as probe pin. pin configuration now looks like this:
> `pins_RUMBA32_common.h` :
> 
> ```
> #define Z_MAX_PIN                           PA15 // PA15 is not connected to anything
> ```
> 
> `pins_RUMBA32_AUS3D.h` :
> 
> ```
> #define SERVO0_PIN                          PB10 // PB10 was Z_MAX_PIN 
> ```
> 
> I am using TMC2209 without any issues now I had problems with 1 drive, which was not running at al but only making noise, I found out this drive was wired different than the other drives which came from the same order... so I swopped 2 wires and it ran.

I assume you are using marlin, right?

May i ask what's your motor degrees ? I have made a setup with klipper. it is easier to configure with klipper.

I can send you my config if you like. I also use use 2209s on uart mode.

**Comment by dixi83** [comment]
Yes it is marlin and I don't have any experience with klipper. The printer I'm building right now is my 5th (2nd CoreXY) custom designed printer. And all the others run/ran marlin as well so I think I'm addicted ;)
My motors are 1.8°/step since we have microstepping I don't see a reason to choose for 0.9° drives. It saves money as well.

**Comment by Makersmic** [comment]
WhiteRabbit62, could you share your printer.cfg file and a pic of your physical connections?  I know it's a lot but I'm having a heck of a time getting these tmc2208's to work in uart mode with my btt rumba32 using Klipper.

**Comment by Cheesus13** [comment]
> It is not the same as the MKS it has many comparable things, for example to get uart drives (like TMC 2208/9) to work, this is what i found after tracing back the board:
> `pins_RUMBA32_AUS3D.h` (which is the pin file BTT uses in there example, the MKS pin file is not even used):
> 
> ```
>     #define X_SERIAL_TX_PIN                 PC14 //PA14 BTT Rumba32 only uses 1 pin for uart
>     #define X_SERIAL_RX_PIN                 PC14
> 
>     #define Y_SERIAL_TX_PIN                 PE4 //PA13
>     #define Y_SERIAL_RX_PIN                 PE4
> 
>     #define Z_SERIAL_TX_PIN                 PE0 //PB10
>     #define Z_SERIAL_RX_PIN                 PE0
> 
>     #define E0_SERIAL_TX_PIN                PC13 //PD11
>     #define E0_SERIAL_RX_PIN                PC13
> 
>     #define E1_SERIAL_TX_PIN                PD5 //PB3
>     #define E1_SERIAL_RX_PIN                PD5
> 
>     #define E2_SERIAL_TX_PIN                PD1 //PB4
>     #define E2_SERIAL_RX_PIN                PD1
> ```
> 
> And I have found more differences and undocumented things. I'll keep this for my self until the lazy company called "bigtreetech" supplied there schematics or unless someone specifically ask for it here. Because the word "support" is not in there dictionary.

Thank you, I finally made UART work. Correct me if I'm wrong, but we can't read from TMC chip using this method? So things like sensorless homing become unavaliable?
What do you think about using free pins from EXP3? We just need a couple of wires and resistors.

**Comment by NebratViacheslav** [comment]
> > I'm using BLtouch for which I redefined the Z-Max pin as servo, and use the Z-Min pin as probe pin. pin configuration now looks like this:
> > `pins_RUMBA32_common.h` :
> > ```
> > #define Z_MAX_PIN                           PA15 // PA15 is not connected to anything
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `pins_RUMBA32_AUS3D.h` :
> > ```
> > #define SERVO0_PIN                          PB10 // PB10 was Z_MAX_PIN 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I am using TMC2209 without any issues now I had problems with 1 drive, which was not running at al but only making noise, I found out this drive was wired different than the other drives which came from the same order... so I swopped 2 wires and it ran.
> 
> I assume you are using marlin, right?
> 
> May i ask what's your motor degrees ? I have made a setup with klipper. it is easier to configure with klipper.
> 
> I can send you my config if you like. I also use use 2209s on uart mode.

Is it posible to use SKR Rumba32 with Klipper? I am asking because can't find this board in "config" Klipper

**Comment by mariokarma** [comment]
> It is not the same as the MKS it has many comparable things, for example to get uart drives (like TMC 2208/9) to work, this is what i found after tracing back the board: `pins_RUMBA32_AUS3D.h` (which is the pin file BTT uses in there example, the MKS pin file is not even used):
> 
> ```
>     #define X_SERIAL_TX_PIN                 PC14 //PA14 BTT Rumba32 only uses 1 pin for uart
>     #define X_SERIAL_RX_PIN                 PC14
> 
>     #define Y_SERIAL_TX_PIN                 PE4 //PA13
>     #define Y_SERIAL_RX_PIN                 PE4
> 
>     #define Z_SERIAL_TX_PIN                 PE0 //PB10
>     #define Z_SERIAL_RX_PIN                 PE0
> 
>     #define E0_SERIAL_TX_PIN                PC13 //PD11
>     #define E0_SERIAL_RX_PIN                PC13
> 
>     #define E1_SERIAL_TX_PIN                PD5 //PB3
>     #define E1_SERIAL_RX_PIN                PD5
> 
>     #define E2_SERIAL_TX_PIN                PD1 //PB4
>     #define E2_SERIAL_RX_PIN                PD1
> ```
> 
> And I have found more differences and undocumented things. I'll keep this for my self until the lazy company called "bigtreetech" supplied there schematics or unless someone specifically ask for it here. Because the word "support" is not in there dictionary.

I really need it 😅
